### PR TITLE
[LANG-1177] Added indexesOf methods and simplified removeAllOccurences

### PR DESCRIPTION
**Pull request by lielfr**
Based of the opened issue. Tests for indexesOf were also added.

### CODE REVIEW

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 1]
```diff
@@ -3254,6 +3254,57 @@ public static int indexOf(final Object[] array, final Object objectToFind, int s
...
 
+    /**
+     * <p>Finds the indices of the given object in the array.
+     *
+     * <p>This method returns an empty BitSet for a {@code null} input array.
+     *
+     * @param array  the array to search through for the object, may be {@code null}
+     * @param objectToFind  the object to find, may be {@code null}
+     * @return a BitSet of all the indices of the object within the array,
+     *  an empty BitSet if not found or {@code null} array input
```
Comment (Stzx) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: `@since` tag should be provided for newly provided methods.

Comment (lielfr) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Okay, which version? 4.0?

Comment (Stzx) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: The next version is `3.10`.

Comment (lielfr) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Done

**Comment by garydgregory** [comment]
@lielfr 
Please rebase on master.

**Comment by lielfr** [comment]
@garydgregory Should be okay now

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 2]
```diff
@@ -8202,20 +8654,8 @@ public static boolean isSorted(final boolean[] array) {
...
-        final int[] indices = new int[array.length - index];
-        indices[0] = index;
-        int count = 1;
-
-        while ((index = indexOf(array, element, indices[count - 1] + 1)) != INDEX_NOT_FOUND) {
-            indices[count++] = index;
-        }
-
-        return removeAll(array, Arrays.copyOf(indices, count));
+        BitSet foundBits = indexesOf(array, element);
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: How about inlining all of the 'foundBits'?

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 3]
```diff
@@ -3254,6 +3254,59 @@ public static int indexOf(final Object[] array, final Object objectToFind, int s
...
+     *
+     * @param array  the array to search through for the object, may be {@code null}
+     * @param objectToFind  the object to find, may be {@code null}
+     * @param startIndex  the index to start searching at
+     * @return a BitSet of all the indices of the object within the array starting at the index,
+     *  an empty BitSet if not found or {@code null} array input
+     * @since 3.10
+     */
+    public static BitSet indexesOf(final Object[] array, final Object objectToFind, int startIndex) {
+        BitSet ret = new BitSet();
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: It seems to me "ret" is not a great name, how about "bitSet"?

Comment (lielfr) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Done

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java (hunk 2 above)
Comment (lielfr) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Done

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 4]
```diff
@@ -3254,6 +3254,59 @@ public static int indexOf(final Object[] array, final Object objectToFind, int s
         return INDEX_NOT_FOUND;
     }
 
+    /**
+     * <p>Finds the indices of the given object in the array.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: Oops, I did not notice before that you did not close any of these HTML tags. You need to close paragraphs.

Comment (lielfr) [reply src/main/java/org/apache/commons/lang3/ArrayUtils.java]: @garydgregory Fixed. Apparently this is a problem with other functions too

**Pull request merged**